web/views.py

```
@@ -6469,6 +6498,61 @@ def all_study_groups(request):
+    # Check if the session is in the past or ongoing
+    now = timezone.now()
+    if session.start_time > now:
+        messages.error(request, "You cannot report attendance for a future session!")
+        return redirect("course_detail", slug=session.course.slug)
+
+    # Check if student has already reported attendance
+    existing_attendance = SessionAttendance.objects.filter(session=session, student=request.user).first()
+
+    if existing_attendance and existing_attendance.status == "pending":
+        messages.info(
+            request, "You have already reported attendance for this session. It is pending teacher verification."
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)